apps/vaporgui/FlowSubtabs.cpp

```diff
@@ -286,313 +348,264 @@ void FlowSeedingSubtab::Update( VAPoR::DataMgr      *dataMgr,
-        catch( const std::invalid_argument& e ) // If not a long number
-        {
-            std::cerr << "bad input: " << pointers[i]->GetEditText() << std::endl;
-            newVal[i] = oldVal[i];
-            pointers[i]->SetEditText( QString::number( oldVal[i] ) );
-            continue;
-        }
-        
-        if( tmp > 0 )   // In the valid range, which is positive here
-        {
-            newVal[i] = tmp;
-            /* std::stol() would convert "383aaa" without throwing an exception.
-               We set the correct text based on the number identified.        */
-            pointers[i]->SetEditText( QString::number( tmp ) );
-        }
-        else
-        {
-            newVal[i] = oldVal[i];
-            pointers[i]->SetEditText( QString::number( oldVal[i] ) );
-        }
-    }
+    std::vector<long> seedsVector(4, (long)1.0);
+    seedsVector[X] = _xSeedSliderEdit->GetValue();
+    seedsVector[Y] = _ySeedSliderEdit->GetValue();
+    seedsVector[Z] = _zSeedSliderEdit->GetValue();
+    seedsVector[RANDOM_INDEX] = _randomSeedsSliderEdit->GetValue();
+    _params->SetRakeNumOfSeeds( seedsVector );
+}
 
-    /* Only write back to _params when newVal is different from oldVal */
-    bool diff = false;
-    for( int i = 0; i < 4; i++ )
-    {
-        if( newVal[i] != oldVal[i] )
-        {
-            diff = true;
-            break;
-        }
-    }
-    if( diff )
-    {
-        _params->SetRakeNumOfSeeds( newVal );
-    }
+void 
+FlowSeedingSubtab::_seedListFileChanged( const std::string& value ) {
+    _params->SetSeedInputFilename( value );
 }
 
+
 void
-FlowSeedingSubtab::_rakeGeometryChanged()
+FlowSeedingSubtab::_rakeGeometryChanged( const std::vector<float>& range )
 {
-    std::vector<float> range;
-    _rake->GetCurrentValues( range );
     VAssert( range.size() == 6 );
     _params->SetRake( range );
 }
 
 void
-FlowSeedingSubtab::_seedGenModeChanged( int newIdx )
+FlowSeedingSubtab::_selectedTabChanged(int index)
 {
-    _params->SetSeedGenMode( newIdx );
-}
+    if (!_paramsMgr)
+        return;
 
-void
-FlowSeedingSubtab::_fileReaderChanged()
-{
-    std::string filename = _fileReader->GetPath();
-    _params->SetSeedInputFilename( filename );
+    const QTabWidget *parent = dynamic_cast<QTabWidget*>(sender());
+    VAssert(parent);
+    const QScrollArea *area = dynamic_cast<QScrollArea*>(parent->widget(index));
+    VAssert(area);
+    const QWidget *widget = area->widget();
+
+    GUIStateParams *gp = (GUIStateParams *)_paramsMgr->GetParams(GUIStateParams::GetClassType());
+
+    gp->SetFlowSeedTabActive(widget == this);
 }
 
 void
-FlowSeedingSubtab::_fileWriterChanged()
+FlowSeedingSubtab::_seedGenModeChanged( int newIdx )
 {
-    std::string filename = _fileWriter->GetPath();
-    _params->SetFlowlineOutputFilename( filename );
-    _params->SetNeedFlowlineOutput( true );
+#warning FlowSeedingSubtab::_seedGenModeChanged() has a hack to get around numeric values representing seed generation modes
```

Comment:
What's this?

Reply:
It's a warning about a hack needed for writing to FlowParams.  I'll remove it, but I'm not a fan of the workaround that's in place.

The seed generation mode in FlowParams is held as a lookup table:
0 - uniform seed distribution
1 - random seed distribution
2 - random seed distribution with bias
3 - list of seeds

Options 1 and 2 are the same thing.  This function is fired off when a combo-box changes index.  The combo-box has indices 0 for uniform, 1 for random, and 2 for list of seeds.  If the user selects options 1 or 2, we have to ++ the index to apply the appropriate value to FlowParams.

Reply:
Why are we making a distinction in the parameter data base between (1) and (2). Random with a bias of zero is (1). Random with a bias != 0 is option 2..

Reply:
> Why are we making a distinction in the parameter data base between (1) and (2).

I believe we're in agreement, that there should be three settings in the params database, not four.  I didn't make any modifications to the params in this PR.  Just the GUI.